source/ingress.go

```diff
@@ -129,7 +133,11 @@ func (sc *ingressSource) endpointsFromTemplate(ing *v1beta1.Ingress) ([]*endpoin
 
 	hostname := buf.String()
 
-	endpoints = addEndpointsFromTargetAnnotation(ing, hostname, endpoints)
+	endpoints = getEndpointsFromTargetAnnotation(ing, hostname)
```

Comment:
I think this does not belong here, endpointsFromTemplate is called only as a last resort to generate an endpoint, i.e. there is no target annotation